tests/components/twitch/test_twitch.py

```diff
@@ -0,0 +1,172 @@
+"""The tests for an update of the Twitch component."""
+from unittest.mock import MagicMock, patch
+
+from requests import HTTPError
+from twitch.resources import Channel, Follow, Stream, Subscription, User
+
+from homeassistant.components import sensor
+from homeassistant.setup import async_setup_component
+
+ENTITY_ID = "sensor.channel123"
+CONFIG = {
+    sensor.DOMAIN: {
+        "platform": "twitch",
+        "client_id": "1234",
+        "channels": ["channel123"],
+    }
+}
+CONFIG_WITH_OAUTH = {
+    sensor.DOMAIN: {
+        "platform": "twitch",
+        "client_id": "1234",
+        "channels": ["channel123"],
+        "oauth_token": "9876",
+    }
+}
+
+USER_ID = User({"id": 123, "display_name": "channel123", "logo": "logo.png"})
+STREAM_OBJECT_ONLINE = Stream(
+    {
+        "channel": {"game": "Good Game", "status": "Title"},
+        "preview": {"medium": "stream-medium.png"},
+    }
+)
+CHANNEL_OBJECT = Channel({"followers": 42, "views": 24})
+OAUTH_USER_ID = User({"id": 987})
+SUB_ACTIVE = Subscription({"created_at": "2020-01-20T21:22:42", "is_gift": False})
+FOLLOW_ACTIVE = Follow({"created_at": "2020-01-20T21:22:42"})
+
+
+async def test_init(hass):
+    """Test initial config."""
+
+    channels = MagicMock()
+    channels.get_by_id.return_value = CHANNEL_OBJECT
+    streams = MagicMock()
+    streams.get_stream_by_user.return_value = None
+
+    twitch_mock = MagicMock()
+    twitch_mock.users.translate_usernames_to_ids.return_value = [USER_ID]
+    twitch_mock.channels = channels
+    twitch_mock.streams = streams
+
+    with patch("twitch.TwitchClient", return_value=twitch_mock):
```

Comment:
Not sure if that is the root cause but you need to specify the exact path to the module that should be patched:
```suggestion
    with patch("homeassistant.components.twitch.TwitchClient", return_value=twitch_mock):
```

Reply:
The path for the mock is correctly defined I think. The ``TwitchClient` is not part of home assistant. Its imported from a dependency. See [here](https://github.com/home-assistant/home-assistant/blob/d355c54eea02122f9ccba2f4320f2d242d054d91/homeassistant/components/twitch/sensor.py#L5)

Reply:
I am not an expert on this but it seems like you need to specify it precisely.
See
https://github.com/home-assistant/home-assistant/blob/48402d49dcc54cdba26e44cdfb45e95e6e3edd15/tests/components/asuswrt/test_device_tracker.py#L32
and 
https://github.com/home-assistant/home-assistant/blob/48402d49dcc54cdba26e44cdfb45e95e6e3edd15/homeassistant/components/asuswrt/__init__.py#L4
You can test it locally and then we will see if it fixes the issue 😄

Reply:
I already tested it and it didn't fixes it. 
The `TwitchClient` class is not in the `homeassistant.components.twitch` package.

It is located in the `twitch` package of the `python-twitch-client` dependency.

Reply:
Could you try it once again with the following? ✌ I think I forgot the `sensor` part.
```python
with patch("homeassistant.components.twitch.sensor.TwitchClient", return_value=twitch_mock):
```

Reply:
It won't work, there is also no `TwitchClient ` in  
`homeassistant.components.twitch.sensor`. 

Inside the `homeassistant.components.twitch` package is the class `TwitchSensor` but that's not the class I want to mock.

Reply:
But you import it and use it inside of sensor.py, see https://github.com/home-assistant/home-assistant/blob/48402d49dcc54cdba26e44cdfb45e95e6e3edd15/homeassistant/components/twitch/sensor.py#L5

Reply:
You might read a little about python dependencies:

http://python-packaging.readthedocs.io/en/latest/dependencies.html

I'm using this library here:
https://python-twitch-client.readthedocs.io/en/latest/

Reply:
Sry that I got rough on you yesterday @springstan. I deeply apologise myself for it.
I just tried it today again and it seems that your clue was correct about the mock specification.

When I mock the package `twitch.client.TwitchClient` its getting my mocks.

Again, I'm deeply sry what happend yesterday here. 😥

Reply:
No hard feelings man 😄 just glad that you got it working now.